pkg/tool/aws-cli/aws-cli.go

```diff
@@ -0,0 +1,177 @@
+package aws_cli
+
+import (
+	"fmt"
+	"github.com/openshift/backplane-tools/pkg/source/aws"
+	"github.com/openshift/backplane-tools/pkg/source/github"
+	"github.com/openshift/backplane-tools/pkg/utils"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// Tool implements the interface to manage the 'aws-cli' binary
+type Tool struct {
+	source *github.Source
+}
+
+func NewTool() *Tool {
+	t := &Tool{
+		source: github.NewSource("aws", "aws-cli"),
+	}
+	return t
+}
+
+func (t *Tool) Name() string {
+	return "aws"
+}
+
+func (t *Tool) Install(rootDir, latestDir string) error {
+	// Pull latest version from GH
+	version, err := t.source.FetchTag()
+	if err != nil {
+		return err
+	}
+
+	var (
+		awsExec           string
+		awsOldInstallDir  string
+		awsBinaryFilepath string
+		url               string
+		fileExtension     string
+		operatingSys      string
+	)
+
+	toolDir := t.toolDir(rootDir)
+
+	if runtime.GOOS == "linux" {
+		// Assign variables for Linux
+		awsExec = "dist/aws"
+		operatingSys = "linux"
+		fileExtension = ".zip"
+		url = "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-" + version + fileExtension
+
+	} else if runtime.GOOS == "darwin" {
+		// Assign variables for macOS
+		awsExec = "aws-cli.pkg/Payload/aws-cli/aws"
+		operatingSys = "darwin"
+		fileExtension = ".pkg"
+		url = "https://awscli.amazonaws.com/AWSCLIV2" + fileExtension
+
+	} else {
+		// Handle unsupported operating systems
+		return fmt.Errorf("Unsupported operating system:", runtime.GOOS)
+	}
+
+	err = os.RemoveAll(toolDir)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	//Download the latest awscli
+	err = os.MkdirAll(toolDir, os.FileMode(0755))
```

Comment:
The current implementation downloads the tool directly to `~/.local/bin/backplane/aws/`; lets add a version directory (making the full path something like `~/.local/bin/backplane/aws/<version>/`) so that we automatically cache old versions of the tool. This gives us something to fall back on if the latest version is broken or has compatibility issues with our internal scripts :slightly_smiling_face: